api/orchestration_images.go

```diff
@@ -42,3 +43,35 @@ func (o *ec2Orchestrator) createImage(ctx context.Context, req *Ec2ImageCreateRe
 
 	return imageId, nil
 }
+
+func (o *ec2Orchestrator) deleteImage(ctx context.Context, id string) error {
+
+	log.Debugf("got request to delete image %s", id)
+
+	input := &ec2.DeregisterImageInput{
+		ImageId: aws.String(id),
+	}
+	if err := o.ec2Client.DeregisterImage(ctx, input); err != nil {
+		return err
+	}
+
+	snapshotinput := &ec2.DescribeSnapshotsInput{
+		Filters: []*ec2.Filter{{Name: aws.String("description"), Values: aws.StringSlice([]string{fmt.Sprintf("*for %s from vol*", id)})}},
+	}
+
+	out, err := o.ec2Client.DescribeSnapshots(ctx, snapshotinput)
+	if err != nil {
+		return err
+	}
+	for _, s := range out {
+		input := &ec2.DeleteSnapshotInput{
+			SnapshotId: (s.SnapshotId),
+		}
+		if err := o.ec2Client.DeleteSnapshot(ctx, input); err != nil {
+			return err
+			//TODO: need to review the return error condition in the loop
```

Comment:
@tenyo do you think returning error inside this for loop is acceptable? Please suggest.

Reply:
Maybe we can just log a Warn and continue, since it's not a critical failure and we want to delete as many of those snapshots as possible. Also, the actual image has already been deleted, so doesn't really make sense to return error here.